auth-foundation/src/main/java/com/okta/authfoundation/credential/Credential.kt

```diff
@@ -94,10 +96,13 @@ class Credential internal constructor(
      * Returns a [Flow] that emits the current [Token] that's stored and associated with this [Credential].
      */
     fun getTokenFlow(): Flow<Token?> {
-        return state.transformWhile {
-            emit(it.token)
-            it !is CredentialState.Deleted
-        }
+        return state
+            .transformWhile {
+                emit(it)
+                it !is CredentialState.Deleted
+            }
+            .dropWhile { it !is CredentialState.Data }
+            .map { it.token }
```

Comment:
@JayNewstrom I ended up going with this instead of a simple `if (isDeleted) return emptyFlow()` because this avoids a minor race condition if the credential is deleted after that block before the actual flow is returned.

It was definitely an edge case that would give a flow with a single null before closing.

Reply:
I don't feel that strongly about accidentally emitting a single element flow of `null`, so I can switch it back if you would prefer the simplicity of that solution over a solution without the race condition